### PR TITLE
The Big Fat Ugly Holoparasite/Guardian Cleanup Pull Request You Ordered (You Didn't Order This, I Got Mad At Wabbajack Code)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Holopara's summoner var is now a weakref
documented SOME STUFF AROUND
wabbajack doesn't have special code to handle holoparas anymore
a lot of code standard stuff. Biggest thing I saw was lack of early returns

## Why It's Good For The Game

Holopara code is shit but it's the kind of shit that doesn't stay in its pen and reaches out into other random bits of the codebase like wabbajack code like what stop

## Changelog

:cl:
code: Cleaned up some parts of holopara code, added some var documentation
refactor: Holoparas use weakrefs instead of strong ones for their summoner
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -53,14 +53,15 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
...
-	var/magic_fluff_string = "<span class='holoparasite'>You draw the Coder, symbolizing bugs and errors. This shouldn't happen! Submit a bug report!</span>"
-	var/tech_fluff_string = "<span class='holoparasite'>BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!</span>"
-	var/carp_fluff_string = "<span class='holoparasite'>CARP CARP CARP SOME SORT OF HORRIFIC BUG BLAME THE CODERS CARP CARP CARP</span>"
-	var/miner_fluff_string = "<span class='holoparasite'>You encounter... Mythril, it shouldn't exist... Submit a bug report!</span>"
+	var/playstyle_string = span_holoparasite("You are a Guardian without any type. You shouldn't exist!")
+	var/magic_fluff_string = span_holoparasite("You draw the Coder, symbolizing bugs and errors. This shouldn't happen! Submit a bug report!")
+	var/tech_fluff_string = span_holoparasite("BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!")
+	var/carp_fluff_string = span_holoparasite("CARP CARP CARP SOME SORT OF HORRIFIC BUG BLAME THE CODERS CARP CARP CARP")
+	var/miner_fluff_string = span_holoparasite("You encounter... Mythril, it shouldn't exist... Submit a bug report!")
```
Comment (tralezab) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:64]: I should do this with the rest, of the holoparasite strings.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -53,14 +53,15 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	var/list/guardian_overlays[GUARDIAN_TOTAL_LAYERS]
 	var/reset = 0 //if the summoner has reset the guardian already
 	var/cooldown = 0
-	var/mob/living/summoner
+	/// weakref to our summon which SHOULD be a living mob.
+	var/datum/weakref/weak_summoner
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:57]: whats wrong with the name summoner? more clear, even if it is a weakref

Comment (tralezab) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm:57]: so summoner can be used for the resolved weak summoner

Comment (JohnFulpWillard) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm:57]: weakrefs generally just have ``_ref`` at the end

**Pull request closed without merging**